src/magic.cpp

```diff
@@ -785,9 +785,23 @@ void known_magic::learn_spell( const spell_type *sp, player &p, bool force )
     }
     if( !force && sp->spell_class != trait_id( "NONE" ) ) {
         if( can_learn_spell( p, sp->id ) && !p.has_trait( sp->spell_class ) ) {
+            std::string trait_cancel;
+            for( const trait_id &cancel : sp->spell_class->cancels ) {
+                if( cancel == sp->spell_class->cancels.back() &&
+                    sp->spell_class->cancels.back() != sp->spell_class->cancels.front() ) {
+                    trait_cancel = string_format( "%s and %s", trait_cancel, cancel->name() );
+                } else if( cancel == sp->spell_class->cancels.front() ) {
+                    trait_cancel = cancel->name();
+                } else {
+                    trait_cancel = string_format( "%s, %s", trait_cancel, cancel->name() );
+                }
+                if( cancel == sp->spell_class->cancels.back() ) {
+                    trait_cancel += ".";
+                }
+            }
             if( query_yn(
-                    _( "Learning this spell will make you a %s and lock you out of other unique spells.\nContinue?" ),
-                    sp->spell_class.obj().name() ) ) {
+                    _( "Learning this spell will make you a %s and lock you out of %s\nContinue?" ),
```

Comment:
```suggestion
                    _( "Learning this spell will make you a %s and lock you out of %s.\nContinue?" ),
```

Reply:
that period is already added. i noticed it in the screenshot and then fixed it, but didn't update the screenshot.

Reply:
(on line 799 above)

Reply:
It would be better to place it here imo. For the translations and simpler code.